### PR TITLE
Allow configuration of RedisBackend's health_check_interval

### PR DESCRIPTION
**Pull request by mrbcmorris**
## Description

👋 Hey folks,

Was working through a reliability issue on our end with our Redis connections and came across the need to update the `health_check_interval` for the underlying Redis connection.  The goal here is a simple implementation that allows reading this value via environment variable (`redis_backend_health_check_interval`) and passing this through as a `kwarg` to the underlying `redis-py` module.

This is my proposed solution to https://github.com/celery/celery/issues/6665.  Please let me know what you think and thank you for the great library.

Cheers,
Chris

### CODE REVIEW

File: celery/backends/redis.py [hunk 1]
```diff
@@ -213,6 +213,7 @@ def __init__(self, host=None, port=None, db=None, password=None,
         socket_connect_timeout = _get('redis_socket_connect_timeout')
         retry_on_timeout = _get('redis_retry_on_timeout')
         socket_keepalive = _get('redis_socket_keepalive')
+        health_check_interval = _get('redis_backend_health_check_interval')
```
Comment (auvipy) [inline celery/backends/redis.py:216]: since this is a new config we should document it as well

**Review by auvipy — CHANGES_REQUESTED** [review]

File: celery/backends/redis.py (hunk 1 above)
Comment (mrbcmorris) [reply celery/backends/redis.py:216]: Thank you for the review.  Newer to this project so I'm not 100% clear on documentation best practices but I took a stab at it by manually updating docs/userguide/configuration.rst in https://github.com/celery/celery/pull/6666/commits/6c7148b6ef00d27297780092371711380c097996.  Please let me know if there's a better way to accomplish this.

File: docs/userguide/configuration.rst [hunk 2]
```diff
@@ -1167,6 +1167,20 @@ Note that the ``ssl_cert_reqs`` string should be one of ``required``,
 ``optional``, or ``none`` (though, for backwards compatibility, the string
 may also be one of ``CERT_REQUIRED``, ``CERT_OPTIONAL``, ``CERT_NONE``).
 
+
+.. setting:: redis_backend_health_check_interval
+
```
Comment (auvipy) [inline docs/userguide/configuration.rst:1172]: version added notation is needed, 5.1 in this case

**Review by thedrow — APPROVED** [review]
LGTM

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
thanks, both of you!

**Pull request merged**